#! /bin/sh /usr/share/dpatch/dpatch-run
## 05_locale.dpatch by Roland Rosenfeld <roland@debian.org>
##
## All lines beginning with `## DP:' are a description of the patch.
## DP: fix locale settings.

Set LC_CTYPE  otherwise we will have strange effects with LC_NUMERIC creating
"," as the decimal separator, which is wrong for postscript code.

@DPATCH@
--- transfig~/fig2dev/dev/genps.c
+++ transfig/fig2dev/dev/genps.c
@@ -42,6 +42,7 @@
 #include "bound.h"
 #include "psencode.h"
 #include "psfonts.h"
+#include <locale.h>
 
 /* for the xpm package */
 #ifdef USE_XPM
--- transfig~/fig2dev/fig2dev.c
+++ transfig/fig2dev/fig2dev.c
@@ -24,6 +24,7 @@
 #include <sys/types.h>
 #endif
 #include <sys/file.h>
+#include <locale.h>
 #include "fig2dev.h"
 #include "alloc.h"
 #include "object.h"
@@ -378,6 +379,7 @@
 	F_compound	objects;
 	int		status;
 
+	setlocale(LC_CTYPE, "");
 #ifdef HAVE_SETMODE
 	setmode(1,O_BINARY); /* stdout is binary */
 #endif
